packages/remix-dev/cli/commands.ts

```diff
@@ -61,7 +61,11 @@ export async function build(
 
   let start = Date.now();
   let config = await readConfig(remixRoot);
-  await compiler.build(config, { mode: mode, sourcemap });
+  try {
+    await compiler.build(config, { mode: mode, sourcemap });
```

Comment:
looks like we intended to have this code pass `onBuildFailure`, and then we can `process.exit(1)` there instead of a try catch

Reply:
ah, also we use a defaultOnBuildFailure handler, I think the process.exit(1) needs to go there.

@mjackson can you help us understand the intent here? Right now it just does `console.error` in the default build failure handler, seems like that's also the spot we'd want to `process.exit(1)` as well.

Reply:
I think that the default on build error is a great place to exit with an error code.
Would you like me to change?